public/js/main.js

```diff
@@ -31,13 +31,42 @@ const state = {
 async function initCamera() {
```

Comment:
Really like the explicit and granular handling of the various errors that can be raised.
We just have to make sure to let the caller (main function) know that something fatal has happened and that it should stop execution of the rest of the routine - now that we're consuming and dealing with the errors raised here, how do you want to bubble up this signal to main? Currently we're returning an undefined value - do you want to make that the signal that something fatal happened in initCamera() or something more explicit?

Reply:
I think, having the errors shown to the user and thrown so `main()` can fail early is better, I've found a way and will see if it plays nicely when running the app.